3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/zambia_irs_data_collector_performance.psql

```diff
@@ -0,0 +1,72 @@
+-- Deploy reveal_irs_zambia_2020:zambia_irs_data_collector_performance to pg
+-- requires: zambia_irs_spray_event
+
+BEGIN;
+
+SET search_path TO :"schema",public;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS zambia_irs_data_collector_performance AS
+SELECT
+  id,
+  plan_id,
+  district_id,
+  data_collector AS data_collector,
+  days_worked,
+  found,
+  sprayed,
+  refused,
+  other_reason,
+  to_char(start_time,'HH24:MI') AS start_time,
+  to_char(end_time,'HH24:MI') AS end_time,
+  to_char(end_time -start_time ,'HH24:MI') AS field_duration,
+  COALESCE(found / NULLIF(days_worked, 0)::DECIMAL, found) AS avg_found,
+  COALESCE(sprayed / NULLIF(days_worked, 0)::DECIMAL, sprayed) AS avg_sprayed,
+  COALESCE(refused / NULLIF(days_worked, 0)::DECIMAL, refused) AS avg_refused,
+  bottles_full AS bottles_full,
+  bottles_accounted AS bottles_accounted,
+  bottles_empty AS bottles_empty,
+  bottles_lostdamaged AS bottles_lostdamaged,
+  bottles_start AS bottles_start,
+  daily_found AS daily_found,
+  daily_sprayed AS daily_sprayed,
+  found_diff AS found_diff,
+  sprayed_diff AS sprayed_diff,
+  bottles_empty / daily_sprayed::DECIMAL AS usage_rate,
+  CASE
+    WHEN (found_diff = 0 AND sprayed_diff = 0 ) THEN TRUE ELSE FALSE
+  END AS data_quality_check
+FROM (
+  SELECT DISTINCT ON (spray_event.plan_id, district_id)
+    public.uuid_generate_v5(
+      '6ba7b810-9dad-11d1-80b4-00c04fd430c8',
+      concat(district_id, spray_event.plan_id)
+    ) AS id,
+    spray_event.plan_id,
+    district_id,
+    data_collector,
+    sum(found) AS found,
+    sum(sprayed) AS sprayed,
+    sum(refused) AS refused,
+    sum(other_reason) AS other_reason,
+    AVG(start_time::time) AS start_time,
+    AVG(end_time::time) AS end_time,
+    DATEDIFF('day', MIN(spray_event.event_date::timestamp), MAX(spray_event.event_date::timestamp)) AS days_worked,
+    sum(daily_summary.bottles_full) AS bottles_full,
+    sum(daily_summary.bottles_accounted) AS bottles_accounted,
+    sum(daily_summary.bottles_empty) AS bottles_empty,
+    sum(daily_summary.bottles_lostdamaged) AS bottles_lostdamaged,
+    sum(daily_summary.bottles_start) AS bottles_start,
+    sum(daily_summary.daily_found) AS daily_found,
+    sum(daily_summary.daily_sprayed) AS daily_sprayed,
+    sum(sprayed) - sum(daily_summary.daily_sprayed) AS sprayed_diff,
+    sum(found) - sum(daily_summary.daily_found) AS found_diff
+  FROM zambia_irs_spray_event AS spray_event
+  INNER JOIN zambia_daily_summary_event AS daily_summary
+    ON daily_summary.location_id = spray_event.location_id
+    AND daily_summary.plan_id = spray_event.plan_id
+    AND daily_summary.event_date = spray_event.event_date
+    AND daily_summary.provider_id = spray_event.data_collector
+  GROUP BY data_collector, spray_event.plan_id, district_id
+) AS subq;
```

Comment:
does this need to be a subquery too?  A little confusing here (to me) generally.

Reply:
The reason i have this as a subquery is because i need to re-use some aggregations several times. As the [days_worked](https://github.com/OpenSRP/opensrp-reveal-datawarehouse/blob/1a05be18284e65949be92f7466d06db263ab33fc/3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/zambia_irs_data_collector_performance.psql#L53) used [here](https://github.com/OpenSRP/opensrp-reveal-datawarehouse/blob/1a05be18284e65949be92f7466d06db263ab33fc/3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/zambia_irs_data_collector_performance.psql#L22-L24)